### ISSUE TITLE
Epic: Phase 5 — Persistent context (Kosh + RTK)

### ISSUE THREAD

**Issue by ankit1057** [ankit1057/sipllm#54]
Combine Kosh (what matters) with RTK (where state lives) into a **persistent context system** that survives across requests and, ideally, session restarts.

### Target behavior
```
request 1 → context/KV created
request 2 → reuse
request 3 → reuse
session restart → restore useful state
```

### Deliverables
- [ ] Session-scoped persistent context store (Kosh cache + RTK KV, coherent lifecycle).
- [ ] Reuse policy: hit/miss accounting across requests.
- [ ] Restore-on-restart of useful state (bounded, validated for correctness).
- [ ] Especially validated on the coding-agent / same-repository workload (system prompt + repo tree + docs + tool defs reused, not recomputed).

### Exit criterion
A repeated-context session shows a measurable **cache/KV reuse rate > 0** with correct outputs, and useful state survives a restart.

Part of the canonical roadmap — parent: #60.

**Comment by ankit1057** [comment]
### Resolution: Phase 5 — Persistent Context (Kosh + RTK) Completed

The persistent context subsystem integrating Kosh context retention with RTK KV persistence has been fully implemented:

1. **Persistent Session Serialization (`include/llm/session.h`, `src/session.cpp`)**:
   - Implements the "SIPS" v1 binary format (`0x53495053`) with 64-bit model identity verification (`model_id`), dimension guards, and sequential I/O.
   - `session_write`: snapshots committed token sequences and resident KV cache tensor data up to `seq_len`.
   - `session_read`: validates file integrity, dimensions, and model compatibility, restoring the KV cache state.

2. **CLI & Runtime Integration**:
   - Supported via `--save-session <path>` and `--load-session <path>`.
   - Seamlessly integrates with cross-turn context reuse (`--reuse`) so that loaded sessions immediately benefit from zero-compute prefix hits across process restarts.

3. **Validation (`tests/test_session.cpp`)**:
   - Verified that cross-process restored logits exactly match in-process forward pass results.
   - Tested model mismatch rejection, missing file handling, and prefix reuse efficiency.

All unit tests pass: `build/test_session` (4/4 tests passing).
